syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/SyndesisExtensionProjectWizard.java

```diff
@@ -84,20 +85,22 @@ private NewSyndesisExtensionProjectMetaData getProjectMetaData() {
 		if (!locationPage.isInWorkspace()) {
 			metadata.setLocationPath(locationPage.getLocationPath());
 		}
-		metadata.setSyndesisExtensionConfig(getSyndesisExtension());
-		metadata.setTemplate(new BasicSyndesisExtensionXmlProjectTemplate());
+		metadata.setSyndesisExtensionConfig(extensionModel);
+		if (extensionDetailsPage.isCustomConnector()) {
+			metadata.setTemplate(new CustomConnectorProjectTemplate());
+		} else if (extensionDetailsPage.isCamelRoute()) {
+			metadata.setTemplate(new CustomStepAsCamelRouteProjectTemplate());
+		} else {
+			metadata.setTemplate(new CustomStepAsJavaBeanProjectTemplate());
+		}
+		Map<String, String> versions = SyndesisVersionUtil.checkSyndesisVersionExisting(metadata.getSyndesisExtensionConfig().getSyndesisVersion(), new NullProgressMonitor());
```

Comment:
as there is a long-running operation with a monitor, this action might be better placed in SyndesisExtensionProjectCreatorRunnable where the progress monitor can be rendered in the wizard.
(can be done in a future iteration)

Reply:
Right. I will move it.